### PR TITLE
Settings improvements

### LINKED ISSUE #1370: Add settings sections
[Figma](https://www.figma.com/design/rR0I4C46je8VwwqCdrpc5U/%F0%9F%93%B1%E2%9A%99%EF%B8%8F-Settings-page?node-id=112-3980&t=nF6kMPxckim5K7Nw-0)

### LINKED ISSUE #1571: Assign sections to settings
These settings need to be put in section:

## Design

[Figma design with section icons](https://www.figma.com/design/rR0I4C46je8VwwqCdrpc5U/%F0%9F%93%B1%E2%9A%99%EF%B8%8F-Settings-page?node-id=155-5457&t=dKp5CKpYosSoXZGX-1)

Change all to sentence case, add the below sections and text changes:

**General**
Language
Health Check - minimum severity: _Sets the minimum Severity for Health check. Anything lower than the selected option will not be displayed in the Health check page._

**Diagnostics and feedback**
Enable telemetry
Minimum logging level

**Advanced**
Show the game files in mod lists (RENAME) - _Show game files as a mod alongside your added mods_
Show the override mod in mod lists (RENAME) - _Show the Override mod, which contains files generated or modified during gameplay that aren't part of any specific mod._

**Developer tools**
Start CLI backend
Log to console

**Experimental - _Not supported_**
Enable unsupported games
Enable multiple loadouts
Enable Xbox game Pass support

**_TO BE REMOVED_**
Text editor theme (remove this as it adds clutter and you can change it where it's relevant from the toolbar anyway)

### Section design with icons

image

### LINKED ISSUE #1606: Telemetry/User data text changes - multiple locations
**Overlay on app startup**
Heading: Help us improve
Body: _Keep as is_
Positive button label: Yes, that's fine
Negative button label: No thanks

**Settings section**
Section name: Privacy

**Setting item**
Heading: Send usage data
Body: Help us improve this app by sending usage data to Nexus Mods. This information is sent anonymously and will never be shared with a 3rd party.

### PR DESCRIPTION
**Pull request by erri120**
- Added sections, resolves #1370.
- Assigned settings to sections, resolves #1571.
- Re-worded and re-organized settings (also closes #1606).
- Can now add links to settings, added one to link to the Privacy site.

### CODE REVIEW

File: src/NexusMods.App.UI/Settings/LoadoutGridSettings.cs [hunk 1]
```diff
@@ -11,20 +11,17 @@ public record LoadoutGridSettings : ISettings
...
+                .WithDescription("Show game files as a mod alongside your added mods.")
                 .UseBooleanContainer()
             )
             .AddPropertyToUI(x => x.ShowOverride, propertyBuilder => propertyBuilder
-                .AddToSection(sectionId)
-                .WithDisplayName("Show Override")
-                .WithDescription("Shows Override in the Mods page.")
+                .AddToSection(Sections.Advanced)
+                .WithDisplayName("Show Override mod")
+                .WithDescription("Shows the Override mod, which contains files generated or modified during gameplay that aren't part of any specific mod.")
```
Comment (Al12rs) [inline src/NexusMods.App.UI/Settings/LoadoutGridSettings.cs:24]: Consider translations

File: src/NexusMods.CrossPlatform/LoggingSettings.cs [hunk 2]
```diff
@@ -81,9 +78,9 @@ public static ISettingsBuilder Configure(ISettingsBuilder settingsBuilder)
                     .RequiresRestart()
                 )
                 .AddPropertyToUI(x => x.LogToConsole, propertyBuilder => propertyBuilder
-                    .AddToSection(sectionId)
-                    .WithDisplayName("Log to Console")
-                    .WithDescription("When enabled, logs will be written to the console as well as the log file.")
+                    .AddToSection(Sections.DeveloperTools)
+                    .WithDisplayName("Log to console")
+                    .WithDescription("Enables the ConsoleTarget (stdout) for all loggers.")
```
Comment (Al12rs) [inline src/NexusMods.CrossPlatform/LoggingSettings.cs:83]: Consider translations

File: src/NexusMods.App.UI/Pages/Settings/SettingsView.axaml.cs [hunk 3]
```diff
@@ -12,25 +13,38 @@ public SettingsView()
...
+                this.BindCommand(ViewModel,viewModel => viewModel.CloseCommand,view => view.CloseButton)
                     .DisposeWith(d);
 
-                this.OneWayBind(ViewModel,
-                        viewModel => viewModel.SettingEntries,
-                        view => view.SettingEntriesItemsControl.ItemsSource)
-                    .DisposeWith(d);
+                this.WhenAnyValue(
+                    view => view.ViewModel!.SettingEntries,
+                    view => view.ViewModel!.Sections
```
Comment (Al12rs) [inline src/NexusMods.App.UI/Pages/Settings/SettingsView.axaml.cs:27]: These are not expected to change right? This should not get triggered on every setting value change for example, right?

File: src/NexusMods.Settings/Services.cs [hunk 4]
```diff
@@ -8,6 +9,40 @@ public static class Services
     public static IServiceCollection AddSettingsManager(this IServiceCollection serviceCollection)
     {
         return serviceCollection
-            .AddSingleton<ISettingsManager, SettingsManager>();
+            .AddSingleton<ISettingsManager, SettingsManager>()
+            .AddSettingsSection(new SettingsSectionSetup
+            {
+                Id = Sections.General,
+                IconFunc = static () => IconValues.Desktop,
+                Name = "General",
```
Comment (Al12rs) [inline src/NexusMods.Settings/Services.cs:17]: Consider translation

File: src/NexusMods.StandardGameLocators/GameLocatorSettings.cs [hunk 5]
```diff
@@ -10,16 +10,13 @@ public record GameLocatorSettings : ISettings
...
         return settingsBuilder
             .ConfigureStorageBackend<GameLocatorSettings>(builder => builder.UseJson())
             .AddToUI<GameLocatorSettings>(builder => builder
                 .AddPropertyToUI(x => x.EnableXboxGamePass, propertyBuilder => propertyBuilder
-                    .AddToSection(sectionId)
-                    .WithDisplayName("(Experimental) Enable Xbox Game Pass support")
-                    .WithDescription("For testing the Xbox Game Pass detection")
+                    .AddToSection(Sections.Experimental)
+                    .WithDisplayName("Enable Xbox Game Pass support")
+                    .WithDescription("Allows you to manage games installed with Xbox Game Pass.")
```
Comment (Al12rs) [inline src/NexusMods.StandardGameLocators/GameLocatorSettings.cs:19]: Consider translations

**Review by Al12rs — APPROVED** [review]
There are some translations that could be made, but would likely create conflicts with the other PRs we have, we can also do these in future prs.

**Comment by Al12rs** [comment]
Here tests are failing

**Review by Al12rs — APPROVED** [review]

**Comment by halgari** [comment]
Wow, this looks *really* good! (haven't looked at the code, but the screenshots look awesome.

**Comment by captainsandypants** [comment]
This is my bad but when we changed the diagnostics section to "Privacy", I should have also updated the icon.
See [updated design](https://www.figma.com/design/rR0I4C46je8VwwqCdrpc5U/%F0%9F%93%B1%E2%9A%99%EF%B8%8F-Settings-page?node-id=155-5457&t=z8a3NQf373myzElF-1). Fair enough if it's too late to change that now though.

**Comment by captainsandypants** [comment]
"Minimum severity" diagnostic setting needs updated text.

[Figma link](https://www.figma.com/design/rR0I4C46je8VwwqCdrpc5U/%F0%9F%93%B1%E2%9A%99%EF%B8%8F-Settings-page?node-id=155-5457)

image

File: src/Abstractions/NexusMods.Abstractions.Games.Diagnostics/DiagnosticSettings.cs [hunk 6]
```diff
@@ -17,14 +17,13 @@ public class DiagnosticSettings : ISettings
...
-        var sectionId = SectionId.DefaultValue;
+        var sectionId = Sections.General;
 
         return settingsBuilder.AddToUI<DiagnosticSettings>(builder => builder
             .AddPropertyToUI(x => x.MinimumSeverity, propertyBuilder => propertyBuilder
                 .AddToSection(sectionId)
-                .WithDisplayName("Minimum Severity")
-                .WithDescription("Set the minimum Severity for Diagnostics. Any diagnostic with a lower Severity will not appear in the UI.")
+                .WithDisplayName("Health Check sensitivity")
+                .WithDescription("Set the minimum severity for diagnostics. You will not be notified for diagnostics with a severity lower than the selected level.")
```
Comment (Al12rs) [inline src/Abstractions/NexusMods.Abstractions.Games.Diagnostics/DiagnosticSettings.cs:26]:
```suggestion
                .WithDescription("Set the minimum severity for Health Check diagnostics. You will not be notified about diagnostics with a severity lower than the selected level.")
```
From review with lausandy

**Review by Al12rs — CHANGES_REQUESTED** [review]

**Review by Al12rs — APPROVED** [review]
@captainsandypants reviewed in call with me and this should be good to merge

**Pull request merged**